src/lib/util/openUrl.js

```diff
@@ -1,7 +1,12 @@
 import { Linking } from 'react-native';
 
-export default function openUrl(url) {
-  if (url) {
+export default function openUrl(url, customCallback) {
+  if (customCallback) {
+    const result = customCallback(url);
```

Comment:
does this callback really need to be called before opening the url, and what is the use case to having that callback return a boolean?

Reply:
yeah sorry I left part of the answer out, which belongs in the sdk/ota repo. 
Basically the callback would look like this: 
```onLinkPress = (url) => {
    if (url) {
      // some custom logic
    }
    // return true to open with `Linking.openURL
    // return false to handle it yourself
    return true
  }

  render() {
    return (
    	<Markdown onLinkPress={this.onLinkdPress}>{copy}</Markdown>```

So if there is a url, I can add additional functionality, such as AWS event, and then returning true proceeds with the opening the link.

Reply:
in checkout flow controller, this is what the markdown component and custom callback will look like
```
_onLinkPress(url: any): boolean {
    if (url) {
      AWSManager.trackEvent('CASH_DISABLED_LOCATION_MODAL_LEARN_MORE');
    }
    return true;
  },

  _renderCustomView(): React.Element<*> {
    return(
      <Markdown style={MarkdownStyle} onLinkPress={this._onLinkPress}>
        Skill-based cash gaming is not yet allowed in your current state/province/country. While you're still able to make a deposit, you'll only be able to play practice matches without a cash entry fee. [Learn More](https://skillz.zendesk.com/hc/en-us/articles/203685899-Who-is-eligible-to-compete-in-Skillz-competitions-).
      </Markdown>
    );
  },
```

I'll create another pr for this

Reply:
makes sense